src/Avalonia.Base/Controls/ResourceDictionary.cs

```diff
@@ -258,6 +258,18 @@ public bool TryGetValue(object key, out object? value)
             return false;
         }
 
+        /// <summary>
+        /// Ensures that the resource dictionary can hold up to <paramref name="capacity"/> entries without
+        /// any further expansion of its backing storage.
+        /// </summary>
+        /// <remarks>This method has no effect when targeting .NET Standard 2.0.</remarks>
+        public void EnsureCapacity(int capacity)
+        {
+#if !NETSTANDARD2_0
+            Inner.EnsureCapacity(capacity);
```

Comment:
Since Inner is lazy-initialized like this:
```
        private Dictionary<object, object?> Inner => _inner ??= new();
```

it would make sense to check if _inner is null first and pass the capacity to the constructor instead.

Reply:
Change made with the constructor working even in .NET Standard (as suggested in #15070).